darts/models/filtering/kalman_filter.py

```diff
@@ -4,150 +4,193 @@
-            self.Q = Q if Q is not None else np.eye(self.dim_x)
-            self.R = R
-            self.H = H
-            self.F = F if F is not None else np.eye(self.dim_x)
             self.kf = None
-            self.kf_provided = False
+            self.dim_x = dim_x
+            self._kf_provided = False
         else:
             self.kf = kf
-            self.kf_provided = True
+            self.dim_u = kf.state_space.u_dim
+            self.dim_x = kf.state_space.x_dim
+            self.dim_y = kf.state_space.y_dim
+            self._kf_provided = True
+            if self.dim_u > 0:
+                self._expect_covariates = True
 
     def __str__(self):
         return 'KalmanFilter(dim_x={})'.format(self.dim_x)
 
+    def fit(self,
+            series: TimeSeries,
+            covariates: Optional[TimeSeries] = None,
+            num_block_rows: Optional[int] = None) -> None:
+        """
+        Initializes the Kalman filter using the N4SID algorithm.
+
+        Parameters
+        ----------
+        series : TimeSeries
+            The series of outputs (observations) used to infer the underlying state space model.
+            This must be a deterministic series (containing one sample).
+        covariates : Optional[TimeSeries]
+            An optional series of inputs (control signal) that will also be used to infer the underlying state space model.
+            This must be a deterministic series (containing one sample).
+        """
+        if covariates is not None:
+            self._expect_covariates = True
+            raise_if_not(series.n_timesteps == covariates.n_timesteps,
+                         'The number of timesteps in the series and the covariates must match.')
+        
+        # TODO: Handle multiple timeseries. Needs reimplementation of NFourSID?
+        self.dim_y = series.width
+        outputs = series.pd_dataframe()
+        outputs.columns = [f'y_{i}' for i in outputs.columns]
+
+        if covariates is not None:
+            self.dim_u = covariates.width
+            inputs = covariates.pd_dataframe()
+            inputs.columns = [f'u_{i}' for i in inputs.columns]
+            input_columns = list(inputs.columns)
+            measurements = pd.concat([outputs, inputs], axis=1)
+        else:
+            measurements = outputs
+            input_columns = None
+
+        if num_block_rows is None:
+            num_block_rows = max(10, self.dim_x)
+        nfoursid = NFourSID(measurements,
+                            output_columns=list(outputs.columns),
+                            input_columns=input_columns,
+                            num_block_rows=num_block_rows)
+        nfoursid.subspace_identification()
+        state_space_identified, covariance_matrix = nfoursid.system_identification(
+            rank=self.dim_x
+        )
+
+        self.kf = Kalman(state_space_identified, covariance_matrix)
+
+
     def filter(self,
```

Comment:
This is an idea for some future PR, but perhaps we could add in here an argument `return_state`, set to `False` by default, which would then make this method return the inferred state `x` instead of the inferred denoised observations `z`. WDYT? I'm assuming there are cases where one is interested in the state (e.g. when the state is lower dimensional and one wants to do anomaly detection).